eventuals/notification.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "eventuals/lock.h"
+#include "eventuals/then.h"
+
+////////////////////////////////////////////////////////////////////////
+
+namespace eventuals {
+
+////////////////////////////////////////////////////////////////////////
+
+// This header file defines a 'Notification' abstraction, which allows
+// an eventual to be able to wait for a single occurrence of a single
+// event. Said another way, the 'Notification' object maintains a
+// private boolean "notified" that transitions to 'true' at most once.
+//
+// This is similar to 'absl::Notification'
+// (https://abseil.io/docs/cpp/guides/synchronization#notification).
+//
+// You can call 'Wait()' on a 'Notification' to wait until the
+// "notified" state is 'true' and 'Notify()' to set the notification's
+// "notified" state to 'true' and notify all waiting eventuals that
+// the event has occurred. This method should only be called once and
+// will do nothing on repeated calls.
+//
+// Note that while 'Notify()' should only be called once, it is
+// perfectly valid to call 'Wait()' multiple times and/or from
+// multiple eventuals -- even after the notification's "notified"
+// state has been set -- in which case those calls will not wait.
+//
+// Note that the lifetime of a 'Notification' requires careful
+// consideration; it might not be safe to destroy a 'Notification'
+// after calling 'Notify()' since other eventuals may have called, or
+// will call, 'Wait()'.
+class Notification final : public Synchronizable {
+ public:
+  Notification()
+    : notification_(&lock()) {}
+
+  ~Notification() override = default;
```

Comment:
Can we omit this line? (I didn't think you needed to explicitly override a default destructor [or any virtual method] when the one you're overriding is non-pure and has the same visibility as your override)